src/modules/user/user.usecase.ts

```diff
@@ -1377,6 +1379,65 @@ export class UserUseCases {
     );
   }
 
+  async upsertAvatar(
+    user: User,
+    avatarKey: string,
+  ): Promise<Error | { avatar: string }> {
+    if (!avatarKey) {
+      throw new BadRequestException('Avatar key required');
```

Comment:
This should be on the controller. The type is not suggesting that it is optional and thus, undefined or null - nor it should do -. It's the controller's responsability to validate this thing.